core/src/main/groovy/noe/tomcat/configure/ConnectorConfiguratorTomcat.groovy

```diff
@@ -124,14 +124,26 @@ class ConnectorConfiguratorTomcat {
   }
 
   private void createNewConnector(Map<String, Object> attributes) {
+    Node connector
+
+    UpgradeProtocolTomcat upgradeProtocol = (UpgradeProtocolTomcat) attributes.remove('UpgradeProtocol')
```

Comment:
Do we really need to handle `UpgradeProtocol` explicitly, isn't it possible to write the code more generic ?

Reply:
This concern also address ConnectorConfiguratorUtils class.

Reply:
To be honest I think we can't. We re taking inside attributes the value UpgradeProtocol which shows us that we re gonna have to make a new node about it later.We re removing it from attributes inside to seed connector with same level attributes and then we re taking whole class of UpgradeProtocol with its attributes and add them inside the new-deeper node

Reply:
In that way is also more transparent our steps of checking and creating of Nodes structure